python/pfs/drp/stella/extractSpectraTask.py

```diff
@@ -8,6 +8,7 @@ class ExtractSpectraConfig(pexConfig.Config):
     useOptimal = pexConfig.Field(dtype=bool, default=True,
                                  doc="Use optimal extraction? "
                                      "Otherwise, use a simple sum of pixels within the trace.")
+    fiberIds = pexConfig.ListField(dtype=int, default=[], doc="If non-empty, only extract these fiberIds")
```

Comment:
I think you could make a case for `None`, but `[]` is at least equally as good. In general I prefer a default of `[]` because you can iterate over it without checking if it's `None`, but because it's an exclusion list here, it doesn't really matter.

Reply:
The other problem with [] is that it's mutable;  you can change the default within the function.  If you want to use an iterable please use a `set`

Reply:
I think you're thinking of using a hardcoded list in a function `def`. This is a different case, and the authors have prevented the bug you're thinking of:
```
>>> from lsst.pex.config import Config, ListField
>>> class Foo(Config):
...     bar = ListField(dtype=int, default=[], doc="")
... 
>>> foo = Foo()
>>> foo.bar
[]
>>> foo.bar.append(1)
>>> foo.bar
[1]
>>> bar = Foo()
>>> bar.bar
[]
```